cloudinit/sources/helpers/azure.py

```diff
@@ -48,6 +48,37 @@ def impl(*args, **kwargs):
     return impl
 
 
+def byte_swapped(previous_id, current_id):
+    """
+    Azure stores the instance ID with an incorrect byte ordering for the
+    first parts. This corrects the byte order such that it is consistent with
+    that returned by the metadata service.
+    """
+    if previous_id == current_id:
+        return None
+
+    def swap_hexstring(s, width=2):
```

Comment:
```suggestion
    def swap_bytestring(s, width=2):
       return ''.join([byte[::-1] for byte in textwrap.wrap(s, 2)])
```
We might be able to replace this entire function with something way simpler

Reply:
I ran swap_bytestring against example id's 

previous = "D0DF4C54-4ECB-4A4B-9954-5BDF3ED5C3B8"
current = "544CDFD0-CB4E-4B4A-9954-5BDF3ED5C3B8"

it returns Nonetype, the current functionality works though.

Reply:
ahh now I understand what intended behavior is, a unit test would have shown this more clearly for me. How about this instead:

```suggestion
    def swap_bytestring(s, width=2):
        dd = [byte for byte in textwrap.wrap(s, 2)]
        dd.reverse()
        return ''.join(dd)
```